newrelic/resource_newrelic_agent_application_browser_integration_test.go

```diff
@@ -38,6 +39,8 @@ func TestAccNewRelicAgentApplicationBrowser(t *testing.T) {
 				),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckNewRelicAgentApplicationBrowserExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "js_config"),
+					testAccCheckJsConfigNestedAttributes(resourceName, "js_config", []string{"init", "info", "loader_config"}),
```

Comment:
Hi @mbazhlekova - these are the changes I've made to the current test case which does two things - 
- check if the resource comprises (exports) an attribute named `js_config`
- since `js_config` is exported as a string that is converted into a JSON using the `jsondecode()` function in HCL, apply a similar mechanism in the test to convert the string into a JSON and iterate through the keys of the JSON to see if `init`, `info` and `loader_config` are present.

I also felt it might be a good idea the `PreCheck` function to `testAccBrowserApplicationsCleanup` as I found a large number of redundant browser applications in our test New Relic account (created by this test) which would be good to be cleaned up before we run this test.

With both of these changes, upon local tests, I find that the test passes.

Please let me know if this works - I'm also wondering if the update made to the current test case is fine, or this would be better as a new test case. Thank you 🙂

Reply:
Adding a cleanup function is a great idea! 👍